joeynmt/embeddings.py

```diff
@@ -53,3 +61,39 @@ def forward(self, x: Tensor) -> Tensor:
     def __repr__(self):
         return "%s(embedding_dim=%d, vocab_size=%d)" % (
             self.__class__.__name__, self.embedding_dim, self.vocab_size)
+
+    #from fairseq
+    def load_from_file(self, embed_path: str, vocab: Vocabulary):
+        """Load pretrained embedding weights from text file.
+
+        - first line will be ignored (vocabulary size and dimension)
+        - each line should contain word and embedding weights
+          separated by spaces.
+        - should be called after initialization!
+
+        Example:
+            2 5
+            the -0.0230 -0.0264  0.0287  0.0171  0.1403
+            at -0.0395 -0.1286  0.0275  0.0254 -0.0932
+
+        :param embed_path: embedding weights text file
+        :param vocab: Vocabulary object
+        """
+        embed_dict = {}
+        # parse file
+        with io.open(embed_path, 'r', encoding='utf-8',
+                     errors='ignore') as f_embed:
```

Comment:
I'm wondering if we could log the number of tokens of the model vocab that were matched, e.g. "found 899 of 1000 (89.9%) tokens of the vocabulary in the pre-trained embeddings", so that if tokenization is different, e.g. contains BPEs, the user might notice a very low number of matches there.

Reply:
That log message might be a bit ambiguous. Let's say `intersection` = num tokens found both in the pretrained file and joeynmt's vocab. 
- len(intersection) / len(vocab size in pretrained file)
- len(intersection) / len(joeynmt's vocab.stoi)

I guess you are suggesting to log the latter, aren't you?

Reply:
Yes, exactly. Perhaps a better formulation: "Found 899 of 1000 (89.9%) tokens of the NMT vocabulary in the pre-trained embeddings, the remaining ones are initialized randomly."